packages/palette/src/elements/PageLoader/PageLoader.tsx

```diff
@@ -0,0 +1,62 @@
+import React from "react"
+import { Box } from "../Box"
+import { ProgressBar } from "../ProgressBar"
+
+interface PageLoaderProps {
+  percentComplete?: number
+  showBackground?: boolean
+}
+
+interface PageLoaderState {
+  progress: number
+}
+
+/**
+ * A top-level loading bar used when transitioning between pages.
+ *
+ * Spec: https://app.zeplin.io/project/5acd19ff49a1429169c3128b/screen/5d7166295b4fca9d4724c13d
+ */
+export class PageLoader extends React.Component<
+  PageLoaderProps,
+  PageLoaderState
+> {
+  state = {
+    progress: this.props.percentComplete || 0,
+  }
+
+  defaultProps = {
+    showBackground: true,
+  }
+
+  currentProgress = this.props.percentComplete || 0
+  step = 0.5
+  interval
+
+  componentDidMount() {
```

Comment:
This looks great. I’m thinking we should consider a 200-400ms delay before showing the loader. Personally I think it’s best if the progress bar isn’t displayed on every page load but only those that aren’t snappy. What do you think?

Reply:
Or maybe this interval achieves this but was this a consideration?

Reply:
Good idea -- I was going to add a light fade in when it appears so I'll work it in then 👍